<a href="https://colab.research.google.com/github/SabrinaDantas/projeto_estatistica_aplicada_pucrs/blob/main/projeto_estatistica_aplicada.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Projeto - Estatística Aplicada



## Fase 1

### 1.1 Leitura do Banco de Dados

A base de dados refere-se a [*Jogadores Registrados na FIFA*](https://raw.githubusercontent.com/filipezabala/pucrs-tecnologo-bd/main/dados/Anexo_Projeto_fifa_world_national_teams_versa%CC%83o_oficial%2020241.csv) , em formado CSV. A leitura será realizada através da URL utilizando a função ***read.csv( )***.

In [24]:
url_fifa <- 'https://raw.githubusercontent.com/filipezabala/pucrs-tecnologo-bd/main/dados/Anexo_Projeto_fifa_world_national_teams_versa%CC%83o_oficial%2020241.csv'
df_fifa <- read.csv(url_fifa, header = TRUE, sep = ';')

### 1.2 Análise da Estrutura

A análise da estrutura de dados é importante para compreender, identificar possíveis problemas, selecionar variáveis relevantes, preparar os dados e planejar a análise. Em R, utiliza-se a função ***str( )***.

In [25]:
str(df_fifa)
# head(df_fifa)

'data.frame':	718 obs. of  30 variables:
 $ id                           : int  158023 153079 211110 201399 226226 199667 212616 216816 183892 231478 ...
 $ name                         : chr  "Messi" "Aguero" "Dybala" "Icardi" ...
 $ full_name                    : chr  "Lionel Andr\xe9s Messi Cuccittini" "Sergio Leonel Ag\xfcero del Castillo" "Paulo Bruno Exequiel Dybala" "Mauro Emanuel Icardi Rivero" ...
 $ overall_rating               : int  94 89 89 87 82 77 77 78 79 79 ...
 $ value_euro                   : int  110500000 64500000 89000000 64500000 30000000 8500000 12000000 15000000 8500000 18000000 ...
 $ wage_euro                    : int  565000 300000 205000 130000 83000 28000 27000 53000 19000 54000 ...
 $ nationality                  : chr  "Argentina" "Argentina" "Argentina" "Argentina" ...
 $ national_team                : chr  "Argentina" "Argentina" "Argentina" "Argentina" ...
 $ club_team                    : chr  "FC Barcelona" "Manchester City" "Juventus" "Inter" ...
 

No banco de dados da FIFA, temos 718 linhas e 30 variáveis que foram classificadas corretamente. Para prosseguir com a análise de possíveis problemas, será verificado a presença de valores ausentes.

In [26]:
sum(is.na(df_fifa))

[1] 0

Não foi localizado a presença de valores ausentes no banco de dados.


### 1.3 Tipos de Variáveis


1. **Qualitativa Nominal:** Nacionalidade - (`nationality`)

2. **Qualitativa Ordinal:**

3. **Quantitativa Discreta:** Reputação em uma escala de 1 (pior) até 5 (melhor) - (`international_reputation.1.5`)

4. **Quantitativa Contínua:** Peso em KG - (`weight_kgs`)





### 1.4 Estatística Descritiva


A estatística descritiva é importante para resumir e descrever características importantes de um conjunto de dados. Em R, podemos usar a função ***summary( )*** para realizar essa tarefa. Antes disso, será realizado um filtro das colunas que são do tipo inteiro.

In [27]:
filter_int <- sapply(df_fifa, is.integer)
summary(df_fifa[,filter_int])

### RETIRAR O ID DO FILTRO

       id         overall_rating    value_euro          wage_euro     
 Min.   : 20801   Min.   :58.00   Min.   :   230000   Min.   :  1000  
 1st Qu.:189831   1st Qu.:72.00   1st Qu.:  3325000   1st Qu.: 10000  
 Median :204626   Median :76.00   Median :  8000000   Median : 27000  
 Mean   :203668   Mean   :76.46   Mean   : 13897695   Mean   : 49560  
 3rd Qu.:222200   3rd Qu.:80.00   3rd Qu.: 17000000   3rd Qu.: 59000  
 Max.   :246387   Max.   :94.00   Max.   :110500000   Max.   :565000  
      age          height_cm     weight_kgs     international_reputation.1.5.
 Min.   :18.00   Min.   :152   Min.   : 59.00   Min.   :1.000                
 1st Qu.:24.00   1st Qu.:170   1st Qu.: 73.00   1st Qu.:1.000                
 Median :26.00   Median :183   Median : 77.00   Median :1.000                
 Mean   :26.57   Mean   :177   Mean   : 77.67   Mean   :1.673                
 3rd Qu.:29.00   3rd Qu.:188   3rd Qu.: 82.00   3rd Qu.:2.000                
 Max.   :37.00   Max.   :203   Max.

### 1.4 Tabela de Frequência


A tabela de frequências é uma ferramenta útil para analisar dados, ajuda a entender a distribuição dos dados, identificar padrões e outliers, validar os dados e tomar decisões informadas.


Para variáveis do tipo caracter (chr), a função ***sapply( )*** é empregada para criar uma tabela contendo apenas os dados desse tipo, excluindo **'name'** e '**full_name'** No entanto, em análises que envolvem um grande número de colunas, pode ser mais eficiente adotar uma abordagem diferente.

In [28]:
sapply(df_fifa[,c("nationality", "national_team", "club_team")], table)

$nationality

          Argentina           Australia             Austria             Belgium 
                 17                  13                  16                  19 
             Brazil            Bulgaria            Cameroon              Canada 
                 23                   3                  17                  19 
              Chile            Colombia      Czech Republic             Denmark 
                 10                  12                  20                  23 
            Ecuador               Egypt             England             Finland 
                  6                   7                  23                  19 
             France             Germany              Greece             Hungary 
                 23                  23                  21                   8 
            Iceland               Italy         Ivory Coast              Mexico 
                 13                  22                  14                  10 
        Nether